nano/node/wallet.cpp

```diff
@@ -1512,6 +1550,121 @@ void nano::wallets::reload ()
+	while (!stopped)
+	{
+		if (!difficulty_reque.empty ())
+		{
+			auto const first (difficulty_reque.begin ());
+			auto now (std::chrono::steady_clock::now ());
+			auto queued (first->first);
+			auto block (first->second);
+			std::shared_ptr<nano::block> block_l (block);
+			bool confirmed (false);
+			difficulty_reque.erase (first);
+			regeneration_lock.unlock ();
+			if (!block)
+			{
+				return;
+			}
+			auto online = node.rep_crawler.total_weight () > (std::max (node.config.online_weight_minimum.number (), node.delta ()));
+			auto transaction (node.store.tx_begin_read ());
+
+			if (((now - queued) >= node.config.work_recalc_interval && online) || node.network_params.network.is_test_network ())
+			{
+				std::unique_lock<std::mutex> lock (node.active.mutex);
+				auto existing (node.active.roots.find (block->qualified_root ()));
+				if (node.active.roots.end () != existing)
+				{
+					//block may not be in existing yet
+					confirmed = existing->election->confirmed.load ();
+				}
+				else
+				{
+					//and so we fall back to ledger confirmation
+					confirmed = node.ledger.block_confirmed (transaction, block->hash ());
+				}
+				lock.unlock ();
+				uint64_t difficulty (0);
+				nano::work_validate (*block, &difficulty);
+				if (!confirmed && node.active.active_difficulty () > difficulty)
+				{
+					block_l = update_work_action (block);
+					lock.lock ();
+					auto existing (node.active.roots.find (block->qualified_root ()));
+					if (node.active.roots.end () != existing)
+					{
+						//block may not be in existing yet
+						confirmed = existing->election->confirmed.load ();
+					}
+					else
+					{
+						//and so we fall back to ledger confirmation
+						confirmed = node.ledger.block_confirmed (transaction, block->hash ());
+					}
+					lock.unlock ();
+				}
+				if (!confirmed)
+				// block could have confirmed while recalculating work with higher difficulty
+				{
+					if (block != block_l)
+					{
+						lock.lock ();
+						auto existing_l (node.active.roots.find (block->qualified_root ()));
+						if (existing_l != node.active.roots.end ())
+						{
+							auto election_l (existing_l->election);
+							if (election_l->status.winner->hash () == block->hash ())
+							{
+								election_l->status.winner = block_l;
+							}
+							auto current (election_l->blocks.find (block->hash ()));
+							current->second = block_l;
+						}
+						lock.unlock ();
+					}
+					node.network.flood_block (block_l);
+					node.active.update_difficulty (*block_l.get ());
+					lock.lock ();
+					auto existing (node.active.roots.find (block->qualified_root ()));
+					if (node.active.roots.end () != existing)
+					{
+						//block may not be in existing yet
+						confirmed = existing->election->confirmed.load ();
+					}
+					else
+					{
+						//and so we fall back to ledger confirmation
+						confirmed = node.ledger.block_confirmed (transaction, block->hash ());
+					}
```

Comment:
This if/else block (incl comments) is identical to one above it. Can it be refactored?